### PR TITLE
Update tag for L1Trigger-L1TMuon to V01-07-00 - [CMSSW_12_4_X]

### PR DESCRIPTION
**Pull request by perrotta**
Move L1Trigger-L1TMuon data to new tag in 12_4_X, see
https://github.com/cms-data/L1Trigger-L1TMuon/pull/23

### CODE REVIEW

File: data/cmsswdata.txt [hunk 1]
```diff
@@ -3,10 +3,10 @@
 #Once a non-default section is empty then cleanup that section and remove its cmsdist/${PACKAGE_TYPE}.file
 #If there is no customization for the packae then remove its .spec and .file
 [default]
+L1Trigger-L1TMuon=V01-06-00
```
Comment (smuzaffar) [inline data/cmsswdata.txt]: @perrotta , shouldn't this be `V01-07-00`?

**Comment by smuzaffar** [comment]
backport of https://github.com/cms-sw/cmsdist/pull/8137

File: data/cmsswdata.txt (hunk 1 above)
Comment (perrotta) [reply data/cmsswdata.txt]: > @perrotta , shouldn't this be `V01-07-00`?

Indeed, thank you @smuzaffar

**Comment by smuzaffar** [comment]
+externals
its a backport of https://github.com/cms-sw/cmsdist/pull/8137 and only contain new data files, so can be safely merged in 12.4.X

**Comment by perrotta** [comment]
+1
- Needed by #39845
- It only includes new data

**Pull request merged**